### PR TITLE
Issue 6534: Git Actions seems to be picking wrong Java version

### LINKED ISSUE #6534: Git Actions seems to be picking wrong Java version
**Describe the bug**
By some reason, our builds changed the Java version. This is the last successful one in `master`:
```
Run ./gradlew jar compileTest checkstyleMain checkstyleTest --parallel 

> Configure project :
Build JVM Version is : 11.0.11 (AdoptOpenJDK 11.0.11+9)
```
And this is what we see today:
```
Run ./gradlew jar compileTest checkstyleMain checkstyleTest --parallel 
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details

> Configure project :
Build JVM Version is : 1.8.0_312 (Temurin 25.312-b07)
```
This breaks the compilation of the project.

**To Reproduce**
Any build performed from now on seems to suffer from this problem.

**Screenshots**
n/a

**Additional information**
n/a

### PR DESCRIPTION
**Pull request by RaulGracia**
**Change log description**  
Explicitly set `actions/setup-java@v2` in the Git Actions configuration file. Also, explicitly set AdoptJDK 11 as the JDK to use.

**Purpose of the change**  
Fixes #6534.

**What the code does**  
Explicitly sets to use AdoptJDK 11 in our builds, as some change in the underlying Git Actions environment could inadvertently switch it to Java 8, which causes compilation errors.

**How to verify it**  
Build must pass.

### CODE REVIEW

**Review by derekm — APPROVED** [review]
This looks good, checks out in build logs.

**Review by sachin-j-joshi — APPROVED** [review]
LGTM

File: .github/workflows/gradle.yml [hunk 1]
```diff
@@ -68,6 +68,10 @@ jobs:
         run: echo Building a '${{ github.event_name }}' for target '${{ github.ref }}'.
       - name: Checkout
         uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
```
Comment (tkaitchuck) [inline .github/workflows/gradle.yml:73]: From the "setup-java" documentation:
> NOTE: Adopt OpenJDK got moved to Eclipse Temurin and won't be updated anymore. It is highly recommended to migrate workflows from adopt to temurin to keep receiving software and security updates. See more details in the Good-bye AdoptOpenJDK post.

So perhaps we should use 'termurin'?

Comment (RaulGracia) [reply .github/workflows/gradle.yml:73]: Good catch, switched to `Temurin`.

**Review by tkaitchuck — APPROVED** [review]

**Pull request merged**